synapse/handlers/directory.py

```diff
@@ -93,7 +95,7 @@ def create_appservice_association(self, service, room_alias, room_id,
         yield self._create_association(room_alias, room_id, servers)
 
     @defer.inlineCallbacks
-    def delete_association(self, user_id, room_alias):
+    def delete_association(self, requester, user_id, room_alias):
```

Comment:
The `requester` is a separate type, which includes a user. Not sure why the functions in this module ask for both. I'm not even sure why the `requester` is needed.